h2o-core/src/main/java/water/rapids/ast/prims/assign/AstRectangleAssign.java

```diff
@@ -128,11 +128,39 @@ private void assign_frame_frame(Frame dst, int[] cols, AstNumList rows, Frame sr
       return;
     }
     // Handle large case
-    throw H2O.unimpl();
+    Vec[] vecs = ses.copyOnWrite(dst, cols);
+    Vec[] vecs2 = new Vec[cols.length]; // Just the selected columns get updated
+    for (int i = 0; i < cols.length; i++)
+      vecs2[i] = vecs[cols[i]];
+    rows.sort();                // Side-effect internal sort; needed for fast row lookup
+    new RowSliceTask(rows) {
```

Comment:
It is dangerous pattern for future code updates. If somebody modifies the context and put there some expensive object and does not check the rest of code.

There are two solutions;
  1) separated static type
  2) a builder method: 
      ```java
        static public RowSliceTask createXXX(int[] rows) { return new RowSliceTask(rows) {
          @Override void mapChunkSlice() { ... }
        }
       ```